### DISCUSSION TITLE
ZJBeny BCP-A2N OCPP EVSE: Update Firmware

### DISCUSSION THREAD

**Discussion by GuzziDoug** [lbbrhzn/ocpp#858] [Q&A]
Hi,
 searched but no answers... How do I update the firmware for my Beny? (Must be done from the OCPP Server, HA in this case)

 fyi, for Beny users, Firmware BCP-A2N-P_SW1_0_26_HW1.bin is recently available.
 The readme attached says: 
Please send the URL ( via "update Firmware.req")  to OCPP EV Charger for updating the firmware.

(Do not use other HTTP server to update, maybe incompatible.):

Household EV charger, Signel phase with cable
http://106.15.78.131/BCP-A2N-P_SW1_0_25_HW1.bin
etc... (end)

so I need to point the server at that address, but unsure how to go about it.

I feel this will be an issue for other EVSEs too, because all the Firmwares will require updating at some point. 
Perhaps a howto may be required? (I last performed an update with Steve, but my Steve server is no longer...

regards, Doug

**Comment by drc38** [comment]
You can use the ocpp firmware update service under developer tools. It should be self explanatory.

**Comment by gnsa14** [comment]
@GuzziDoug,
Looks like I have same setup as you.
Interestingly i had issue arrise yesterday where my OCPP ZJBeny no longer obeys my Maximum Amps setpoint. It's been working for a month now perfectly. I was on 1.0.24 but have upgraded successfully to 26 as per @drc38 post above. Still no change. Reboot of ZJBeny and HA, updates of HA to Home Assistant 2023.7.1, Supervisor 2023.07.1, Operating System 10.3, Frontend 20230705.1 still no joy.
Looking through the logs, the limit value (22A) is sent and an accepted response is given see below.... Ideas?
Maybe an silent update (other than my forced updates mentioned above) has broken something?

2023-07-07 13:11:50.511 INFO (MainThread) [ocpp] zjbeny: send [2,"2ef99948-786e-4efd-8141-fdea6214398e","SetChargingProfile",{"connectorId":0,"csChargingProfiles":{"chargingProfileId":8,"stackLevel":10,"chargingProfileKind":"Relative","chargingProfilePurpose":"ChargePointMaxProfile","chargingSchedule":{"chargingRateUnit":"A","chargingSchedulePeriod":[{"startPeriod":0,"limit":22.0}]}}}]

2023-07-07 13:11:50.703 INFO (MainThread) [ocpp] zjbeny: receive message [3,"2ef99948-786e-4efd-8141-fdea6214398e",{"status":"Accepted"}]

**Reply by GuzziDoug** [reply]
Hi,
 I only just managed to get the Beny updated. There is a (Revised on  1/7/23?) file on their download site that successfully loaded 1.0.26.
 Before, it refused to download.
 As I am retired, I just plug the car in when the Solar is cranking. Yet to set up charge management. I have Victron M+2 with 12 Kw Lithium Titanate battery (works well), & 15Kw of PV feeding 2 of Solar Edge SE5000. We generate 80Kw/day in summer & abt 40Kw now in winter.
 With 1.0.15, I had problems with occasional lockups. Hopefully fixed in 1.0.26. (Reset by powering off Beny).

**Reply by GuzziDoug** [reply]
Hi,
 I checked this morning: the charger started OK, but as you said, I could not adjust the current. It runs at 5.1Kw no matter where the current limit set. Tried disconnecting & reconnecting (with current set to 15A) but still supplying 5.1`Kw. Also the Rfid does not turn off the charger, neither does the reset (small button). The HA turn off command from the computer works. All the log entries are saying the commands are sent. I feel it might be a firmware issue: I will email Beny.

**Reply by gnsa14** [reply]
Good to know it's not a problem with my hardware!
The "Charger Current Offered" value is the one i see not changing. 
Strange thing is, it worked perfectly since installation running 1.0.24 and stopped without me changing anything. 
I also noticed that the OCPP heartbeat back to HA stopped at the same time the Max setpoint stopped having any effect.

Very interested to hear what responses you get from Beny email.

**Reply by TTMK1994** [reply]
Using this configuration to configure the Charger can correctly limit the current, provided that the Charger is not configured with a higher priority ChargingProfile

connectorId:0
stackLevel:10
chargingProfileld:27
chargingProfileKind:Relative
chargingProfilePurpose:ChargePointllaxProfile
recurrencyKind:Daily
validFrom:2023-07-01T00:00:00.000Z
validTo:2050-12-22T00:00:00.000Z
duration:86400
startSchedule:2023-07-07T00:00:00.000Z
chargingRateUnit:A
minChargingRate:0.0
limit[0]:22.0
startPeriod[0]:0

**Comment by GuzziDoug** [comment]
Thank you for your replies. The Beny charger was working correctly until the upgrade 1.0.26 was applied.
 With firmware 1.0.15 everything worked correctly:
Charger current could be changed while the vehicle was charging from the Home Assistant (HA) console.
The RFID was able to initiate charging, & stop charging when the RFID was used. I understand that the RFID communicates to the server, which then initiates command.
The charger could be started & stopped from the HA Console.

 Since Firmware 1.0.26 was applied:
Charge current inoperable (Console current changes, but charger is delivering 5.1Kw constantly)
RFID will initiate charger start, but then RFID will not stop charger (so communication charger to server is not correct).
Charger can be started & stopped from the HA Console.
It has been noted that the HA server is not receiving the Heartbeat tick signal from the charger.

Information on HA OCPP server:
 The OCPP server (that I supplied a link to) module in HA is OCPP 1.6 compliant, & as stated worked until firmware 1.0.26. I prefer to use HA because it is a total building management package. I read all my data from my PV, Battery, loads, as well as other data. This is not a normal setup, but a farm.
 Can I request a copy of the Firmware change log between ver 1.0.15 & 1.0.26? 

 Can you send me links to the current documentation please. (Why does Beny not put this documentation in a publicly accessible area? It is difficult to find upgraded information.) I will try reconfiguring my charger then enable full error logs on my server. Thank you for the information on the command priorities: I will pass that to the software developers.

 To my knowledge, Z-box & the server were not available with firmware 1.0.15. I still wish to use the HA server, so need to find what has changed so the HA server software can correctly communicate with the Beny charger (this may be due to the HA software not fully complying with OCPP ver 1.6, or Beny....)
 The Home Assistant community is huge. Beny chargers must work with this package. Because the OCPP module 
is Open Source, there is no reason it could not be tested, & if necessary changes suggested. I am not a developer, but I do run much Open Source software. I understand the problems.

 One question regarding Wifi communication to the EV charger: Has the WIFI security been upgraded with 1.0.26 compared to 1.0.15? (at present I use a separate WIFI AP to communicate to the insecure EV charger) 

Thank you, Doug Foskey.

**Reply by gnsa14** [reply]
@GuzziDoug,
I think i stumbled across a fix for this. Mine is now working again!
I sent a command via HA Services to "Clear Charging Profiles"
```
service: ocpp.clear_profile
    data: {}
```
This fixed the Maximum Current setpoint issue.
See how you go with yours.

**Reply by GuzziDoug** [reply]
@gnsa14
Thanks. That  "Clear Charging Profiles" fixed mine too.

 I wonder if the OCPP module developers could add that routine in after a firmware update? It is not unknown for the config data to get corrupted by a firmware update. That also probably applies to other chargers too.

In the meantime, I will mark this Beny Issue solved, & let Beny know too.

**Comment by GuzziDoug** [comment]
Hi,
 we seem to have a solution to the errors caused by upgrading to 1.0.26 firmware.
Reported by fellow user: 

@GuzziDoug,
I think i stumbled across a fix for this. Mine is now working again!
I sent a command via HA Services to "Clear Charging Profiles"

service: ocpp.clear_profile
    data: {}
This fixed the Maximum Current setpoint issue.
See how you go with yours.

(end)

 As a suggestion perhaps the charger could perform a clear profile after a Firmware update? This would mean the operator would not see the problem, so increase reliability.

regards, Doug

ps: can you send me links to the current documentation please. I am hoping WIFI security has been upgraded since 1.0.14.

**Comment by gnsa14** [comment]
For those interested, there was a new OCPP firmware released at the end of July 2023

"Enhanced compatibility with OCPP (Open Charge Point Protocol) charging station firmware version 1.0.27, introducing additional functionalities such as: energy limitation charging – one-time, scheduled charging, weekly reservations, historical energy consumption, and monthly charging restrictions."
I took this from the Beny app page, so there may be some new information available via OCPP also.
I am yet to try this myself.

**Comment by PeterBorfai** [comment]
I'm looking and buying a Beny EVSE but first I need to test it. The retailer is having difficulty in configuring the CP to connect to my Home Assistant. Can anyone suggest the necessary URL format?  I've also tried testing from https://evspace.in/cp-simulator but I don't see how the path and the ID should look. Hope someone can clarify this for me. Thanks in advance.

**Comment by lbbrhzn** [comment]
For a home assistant installation with public ip address 'homeassistant.domain' and OCPP installed on port 9000, the URL should look like this:

`wss://homeassistant.domain:9000/MY_CHARGER`

To use the secure websocket protocol (wss:) you need to upload the SSL certificates to the home assistant installation, and indicate the path to the .pem files in the OCPP configuration flow. You may also need to forward this port on your router. For a private connection through the local network this is not required, you can use the regular websocket protocol (ws:) instead.

The charge point id 'MY_CHARGER' and the path is currently ignored by the OCPP integration. But in the future this may change, in order to allow you to set a password per charge point.

**Reply by GuzziDoug** [reply]
Peter,
 I am no expert, but the OCPP ZJBeny has been working with no problems. My only issue has been I have an early hardware version, & the WiFi security is woeful. Not a problem for me as I am in a rural area. The later hardware has ethernet capability which I feel would be better. The Rfid works fine & I use it to turn the charger on & off. The H/A OCPP driver works well & the charging sessions  are logged & could be better aggregated if you write a routine (if you need to have separate logging for vehicles for instance).
 The charger firmware needs occasional updating: I do this manually, pointing to the Beny address, then perform a ´Clear Charging Profiles'after the upgrade is successful (as mentioned in an earlier post). All in all, an economical charger that works well.

**Reply by PeterBorfai** [reply]
Thanks lbbrhzn for your clarifications. The simulator is connecting now! And thanks to GuzziDoug for your comments. I'm planning to test with the retailer this week, but based on your comments I have confidence this low-cost option will work!

**Comment by Idan37S** [comment]
Question to all of you please about ZJBeny charger - 
Is it possible to use both OCPP with HA and the z-box app simultaneously?

**Comment by GuzziDoug** [comment]
I have the ZJBeny OCPP, & it came with the z-box. I never bothered using it because 
it was not needed. It is far better to write some routines to control the charger o/p 
from HA, although I must admit I have never done that! (I manually change the current 
based on my needs).
 One reason I did not use the z-box was that I would have required more sensors on my 
system, & my generation is 40M from my house/loads. Also the early s/w was allegedly 
not that reliable.

regards, Doug
ps: when you write some routines, I might use them to automate my system!

**Comment by Idan37S** [comment]
Thanks @GuzziDoug, I will share some once I will have my charger :)

Can someone please share what entities are there with this charger and OCPP in HA?
I'm just about to buy it and want to make sure I have everything I need there.

**Reply by gnsa14** [reply]
Here is the full list in HA of what you have access to.  If only the SoC came through too...
![ocpp](https://github.com/user-attachments/assets/b44fb6b5-2bee-41c3-bdde-8b02910a921f)

**Reply by Idan37S** [reply]
@gnsa14 What are the Availability switch and the Charge Control switch?

**Reply by gnsa14** [reply]
Availability on = charge plug in charger (not being used, therfore available)
Charge Control Switch is the turn on/off charging switch, so you can have it plugged in, and schedule the charging via HA for example

**Reply by gnsa14** [reply]
I use one of the commands in HA to update the firmware too.

**Comment by GuzziDoug** [comment]
If you look higher up I have detailed how to upgrade the firmware, inc 
the address of the ZJB download page.

**Comment by Idan37S** [comment]
I have connected the charger to HA via OCPP, and at first everything was working great,
But, now all of a sudden I **can't change the Max Current anymore**, whenever I try to change it I get a **"Protocol error"** and it goes back to the previous value.

I have tried clearing charging profile without luck
```
service: ocpp.clear_profile
    data: {}
```
In the logs I see this:
```
2025-04-30 09:49:04.551 INFO (MainThread) [ocpp] 240300155: send [2,"73986dd1-09cc-4713-8028-cd3c769f03be","ClearChargingProfile",{}]
2025-04-30 09:49:04.648 INFO (MainThread) [ocpp] 240300155: receive message [3,"73986dd1-09cc-4713-8028-cd3c769f03be",{"status":"Unknown"}]
2025-04-30 09:49:04.649 WARNING (MainThread) [custom_components.ocpp] Failed with response: Unknown
```

I also tried removing and re-adding the integration and charger, tried resetting the charger, but nothing works.
I have tried with the option **“Force Smart Charging feature profile”** enabled and disabled during setup of the device in HA if that matter.

**Anyone has any idea how to resolve it?**
I'm using **FW v1.0.39 (model BCP-AT2N-P)**, and these are the errors in the logs:

```
2025-04-30 09:21:06.338 INFO (MainThread) [ocpp] 240300155: send [2,"7e7141b3-85ad-455f-805a-24b83c4f7293","GetConfiguration",{"key":["ChargingScheduleAllowedChargingRateUnit"]}]
2025-04-30 09:21:06.636 INFO (MainThread) [ocpp] 240300155: receive message [3,"7e7141b3-85ad-455f-805a-24b83c4f7293",{"configurationKey":[{"key":"ChargingScheduleAllowedChargingRateUnit","readonly":true,"value":"Current,Power"}]}]
2025-04-30 09:21:06.636 DEBUG (MainThread) [custom_components.ocpp] Get Configuration for ChargingScheduleAllowedChargingRateUnit: Current,Power
2025-04-30 09:21:06.636 INFO (MainThread) [custom_components.ocpp] Charger supports setting the following units: Current,Power
2025-04-30 09:21:06.636 INFO (MainThread) [custom_components.ocpp] If more than one unit supported default unit is Amps
2025-04-30 09:21:06.637 INFO (MainThread) [ocpp] 240300155: send [2,"0da985f2-532d-492c-b89d-4fdd3988f144","GetConfiguration",{"key":["ChargeProfileMaxStackLevel"]}]
2025-04-30 09:21:08.743 INFO (MainThread) [ocpp] 240300155: receive message [3,"0da985f2-532d-492c-b89d-4fdd3988f144",{"configurationKey":[{"key":"ChargeProfileMaxStackLevel","readonly":true,"value":"10"}]}]
2025-04-30 09:21:08.744 DEBUG (MainThread) [custom_components.ocpp] Get Configuration for ChargeProfileMaxStackLevel: 10
2025-04-30 09:21:08.744 INFO (MainThread) [ocpp] 240300155: send [2,"b2641958-02b6-43be-9a60-d017393b853e","SetChargingProfile",{"connectorId":0,"csChargingProfiles":{"chargingProfileId":8,"stackLevel":10,"chargingProfileKind":"Relative","chargingProfilePurpose":"ChargePointMaxProfile","chargingSchedule":{"chargingRateUnit":"A","chargingSchedulePeriod":[{"startPeriod":0,"limit":10.0}]}}}]
2025-04-30 09:21:10.770 INFO (MainThread) [ocpp] 240300155: receive message [4,"b2641958-02b6-43be-9a60-d017393b853e","ProtocolError","",{"errorMsg":""}]
2025-04-30 09:21:10.770 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [23361145140512] Unexpected exception
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 245, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<7 lines>...
    )
    ^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2794, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2837, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1007, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
        hass, entity, func, data, call.context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1079, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/number/__init__.py", line 122, in async_set_value
    await entity.async_set_native_value(native_value)
  File "/config/custom_components/ocpp/number.py", line 131, in async_set_native_value
    resp = await self.central_system.set_max_charge_rate_amps(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        self.cpid, num_value
        ^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/config/custom_components/ocpp/api.py", line 345, in set_max_charge_rate_amps
    return await self.charge_points[cp_id].set_charge_rate(limit_amps=value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ocpp/ocppv16.py", line 323, in set_charge_rate
    resp = await self.call(req)
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/ocpp/charge_point.py", line 421, in call
    response = await self._get_specific_response(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        call.unique_id, self._response_timeout
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/config/custom_components/ocpp/chargepoint.py", line 341, in _get_specific_response
    raise resp.to_exception()
```

**Comment by GuzziDoug** [comment]
as a long time user of the zjbeny, everything just worked!
 But there was one occasion that required me to re-setup the 
HA page (but I cannot remember the reason!).
 The latest firmware is ver Firmware: 1.0.39. If req search back & 
find how & where to find the firmware. Remember to power cycle the 
charger once the firmware is fully updated (& it does not hurt to 
clear profile too).
 One thing I have found on the ZJB firmware page is the config is not always 
pointing to the latest firmware. I edit the HA load to point to the 
latest firmware (but ensure you load the correct config!)

Good luck, Doug

**Reply by Idan37S** [reply]
I believe the root problem is that clearing the profile is failing in my case, and i'm not sure why or how to resolve it.
I have tried power cycle (restarting the charger) and also factory reset and reconnecting it again to the ocpp ha server,
But it didn't help.

Did anyone encountered it?
Is there another way to clear charging profiles?
Does anyone has a way to contact the manufacture?
```
2025-04-30 09:49:04.551 INFO (MainThread) [ocpp] [charger]: send [2,[key],"ClearChargingProfile",{}]
2025-04-30 09:49:04.648 INFO (MainThread) [ocpp] [charger]: receive message [3,[key],{"status":"Unknown"}]
2025-04-30 09:49:04.649 WARNING (MainThread) [custom_components.ocpp] Failed with response: Unknown
```

**Comment by Idan37S** [comment]
For anyone who will encounter my issue, 
Disabling **OCA Mode** in the charger has fixed all my issues and everything is working great now.

**Reply by zak59950** [reply]
Hello, how do I disable OCA mode?

**Comment by Idan37S** [comment]
Adding some HA cool stuff that I have done with my new charger:
- Charger Card using this great HACS card: https://github.com/tmjo/charger-card
   - Code: https://pastecode.io/s/rce0fp9p
   - Screenshots:
![available](https://github.com/user-attachments/assets/d7a1adfe-bc71-4aaa-8e79-7dd2b8a45a43)
![charging](https://github.com/user-attachments/assets/e47adcb6-36ea-4963-a513-6b36b9376e3d)

- Automation: https://pastecode.io/s/i9nytn00
   - The automation is mainly for automatic start charging when plugged, stop charging when done, and telegram notifications

**Comment by zak59950** [comment]
Hello, how do I disable OCA mode?

**Reply by Idan37S** [reply]
> Hello, how do I disable OCA mode?
When you enter web configuration mode (see instructions here: https://ultipower.com.au/pages/ocpp-service-zj-beny-charger-network-configuration-guide?srsltid=AfmBOoonTrhfN2zGdfQjs-jPNoUTQPBU6wYUV6cwbxRexW5aXt33jCkU),
You can see that config of OCA mode.

**Comment by GuzziDoug** [comment]
I never had to worry about OCA. Another comment said it was in the initial 
config when you log in with a phone, etc for the initial config.

**Reply by zak59950** [reply]
Hello and thank you.
I can't get into AP mode.
My base station is BCP-B2N-L.
I don't understand.

**Comment by GuzziDoug** [comment]
From the manual:

The method for the charger to enter AP configuration mode:
Ÿ Disconnect the charger.
Ÿ Press the emergency stop button.
Ÿ Press and hold the leakage test button and do not release it.
Ÿ Power on the charger.
Ÿ Release the leakage test button and emergency stop button after the charger beeps.
When the yellow light in the LED flashes, the charger has successfully entered AP configuration mode.

nb: to release the emergency stop, the button is rotated.
You should see the AP come up on wifi on your device eg phone.

One thing when you configure the device on your router, is the security is not strong. Best set it on a 
separate interface & limit it to that MAC address. (I use OpenWRT & it is easy to set up a special interface)
regards, Doug

**Comment by jafar-atili** [comment]
My BCP-AT2N-L was upgraded to 1.1.0 and most entities are unavailable.

I wasn't able to get it connected properly to HA, does anyone know what might be the problem?

![image](https://github.com/user-attachments/assets/dc77a336-d52f-4ab8-9d5d-b28e9407b488)

**Reply by Pseudophryne** [reply]
Thanks for sharing - I shall delay my upgrade 😃

**Comment by jafar-atili** [comment]
> My BCP-AT2N-L was upgraded to 1.1.0 and most entities are unavailable.
> 
> I wasn't able to get it connected properly to HA, does anyone know what might be the problem?
> 
> ![image](https://github.com/user-attachments/assets/dc77a336-d52f-4ab8-9d5d-b28e9407b488)

Downgrading the integration to 0.7.0 fixed the problem

**Comment by GuzziDoug** [comment]
I am running 1.0.39 with no issues. I will wait for 1.1.1 I think? (Next upgrade).

 For those interested I asked ZJB when a bi-directional charger will be avail: 
the reply indicated it is in the pipeline. (Want one!)

**Comment by TTMK1994** [comment]
You can try using the ChangeConfiguration feature to reset the MeterValuesSampledData and MeterValuesAlignedData configuration items




---Original---
From: "Jafar ***@***.***&gt;
Date: Sat, May 24, 2025 19:07 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [lbbrhzn/ocpp] ZJBeny BCP-A2N OCPP EVSE: Update Firmware(Discussion #858)




 
My BCP-AT2N-L was upgraded to 1.1.0 and most entities are unavailable.
 
I wasn't able to get it connected properly to HA, does anyone know what might be the problem?
 
2D5E7E29-0A65-4A29-A103-1BCCBC9C9C7A.png (view on web)

**Comment by Pseudophryne** [comment]
Hi all, a quick question maybe related to ZJ Beny only and before I post in the broader forum.  Has anyone upgraded to HA Core 2025.6.2?   I upgraded today to 2025.6.2 and all entities became unavailable - reverted back to 2025.5.3 and all entities became available immediately.

**Comment by GuzziDoug** [comment]
I upgraded HA core today & seem to have no issues. I am running ZJBeny firmware 1.2.0 (upgraded yesterday). 
Did you clear protocols? I have seen your issue previously when the settings were not cleared. (But yesterday 
the command failed execution, but everything is working so not sure!)
 My BCP-A2N P is a really early one: ser # 22060001! 

regards, Doug (Australia)

**Reply by Pseudophryne** [reply]
Thanks Doug, ocpp.clear_profile was about the only thing I didn't try and I should have thought of that 😃   Even better you answered another question I have about the firmware as I saw 1.2.0 is released.

**Reply by Pseudophryne** [reply]
@GuzziDoug I attempted ocpp.clear_profile and the command will not run in 2025.6.2 = I reverted back to 2025.5.3 and all is well again.   Any other advice anyone can give on this?

**Comment by GuzziDoug** [comment]
Same issue: Clear profile not working & I cannot control current delivered to EV.

regards, Doug

**Reply by Pseudophryne** [reply]
@GuzziDoug I found downgrading OCPP to v0.8.3 fixed everything - seems a number of people are experiencing this issue (or similar) using OCPP v0.8.4, issue 1641 mentions possible cause.

**Reply by gnsa14** [reply]
Ok, so I upgraded to 0.8.4 yesterday.  I was coming from 0.5.3 so I thought it was time. 
I upgraded, and got migrated error again.  I e/stopped and restarted the charger. Then I power cycled the charger. Then I tried rebooting HA. (Not restarting HA)
Still not working.
Then I removed ocpp from HA.
Then I rebooted again. 
Then reinstalled ocpp. May have rebooted again. 
Added discovered harger again,  and it worked.
Then cleared profiles for the fun of it.

Hope that works for you.(or latter part of it)
BTW: Still on 0.39 charger firmware, 2025.06.2 HA supervisor
2025.6.3 HA Core

**Reply by Pseudophryne** [reply]
I should add precision to my comment I found downgrading OCPP to v0.8.3 fixed everything - seems a number of people are experiencing this issue (or similar) using OCPP v0.8.4 with all sorts of chargers and later versions of HA starting a few weeks back, issue #1641 mentions a possible cause.  Note my issue occurred when HA Supervisor upgraded to 2025.06.2 everything was perfect prior to that.

**Comment by Pseudophryne** [comment]
Good day all, I notice firmware update BCP-A2N-P_SW1_2_6_HW1.bin is released, has anyone updated their firmware to v1.2.6 and what is your experience post update to v1.2.6?

**Reply by GuzziDoug** [reply]
Just downloaded & loaded: Also running latest HA core. Remember to power cycle charger after update.
All seems OK so far (current & OCPP control on HA seem to be working OK.)
 Did anyone look at the change notes?

**Reply by GuzziDoug** [reply]
OK, looked at the change notes: Security updates to allow secure login (not before time!)
 & changes to DLB box support for those that use one, that also seem useful (inc neg Energy pricing support!).
 They also now have an option for Temp sensing the Vehicle plug, which seems worthwhile I feel (but my 
charger is a really early one, so not applicable)

**Reply by Pseudophryne** [reply]
Thanks for your feedback, ironically I updated about 30mins back after reading the release notes and decided the risk was low and benefits for security outweighed the risk.   All is working well after the firmware upgrade to v1.2.6 :smile:

**Reply by painkillerluo** [reply]
Sounds like those security enhancements only apply to the BCPC model. C for Cyber security maybe. Updated my BCP. Seems ok.

**Reply by imjordanxd** [reply]
Where have you managed to download it from? It's so difficult to this these anywhere

**Reply by Pseudophryne** [reply]
> Where have you managed to download it from? It's so difficult to this these anywhere

All firmware from Beny is on their download site http://106.15.78.131

**Comment by chiefcomm** [comment]
Hey all, quick question anyone tried Open Charge Point Protocol (OCPP) v0.10.8 with ZJBeny BCP-A2N? I've stayed on v0.9.0 with v1.2.6 FW to date and all works well any experiences with the newer versions of OCPP and FW?

**Comment by chiefcomm** [comment]
Hi all you Beny users, wondering if any of you are experiencing issues with changing the Maximum Current in HA?   This was working flawlessly way back in OCPP v0.9.0 but since upgrading to OCPP v0.10.x I've been having no joy in changing the Maximum Current.   I have done all the normal tricks ocpp.clear_profile removing the addon completely and starting again.   I have Beny BCP-A2N-P and other than charging current it works flawlessly with the addon.  Thoughts?

**Reply by chiefcomm** [reply]
Well, you wouldn't believe it. I spent a couple of hours this morning and finally all is working well but it was a saga.   I deleted OCPP and purged all entities with a few restarts.  Added OCPP app back and added the charger again with a different name and it started to work perfectly (go figure). Since then I've renamed all the entities to maintain my history data and all is wonderful with my ZJ Beny BCP-A2N-P and Home Assistant

**Comment by ToastySefac** [comment]
Yeah, I had the same thing last week. Had to remove everything and start again. Was stuck at 18A max.
